### PR TITLE
Prepare release 4.0.0 and bump version to 4.1.0a1

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -44,17 +50,10 @@ Fixed
   process is finished
 - Remove unrelated permissions occasionally listed among group
   permissions on ``permissions`` endpoint
-- Fix ``ResolwePermissionsMixin`` to work correctly with multi-words 
-  model names, i.e. DescriptorSchema
-
-
-==================
-3.0.1 - 2017-09-15
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Why was this deleted? You should also include change logs for all `3.x` versions.

File: docs/CHANGELOG.rst [hunk 2]
```diff
@@ -44,17 +50,10 @@ Fixed
...
-==================
-
-Fixed
------
-- Correctly relabel SELinux contexts on user/group files
+- Fix ``ResolwePermissionsMixin`` to work correctly with multi-words
+  model names, i.e. ``DescriptorSchema``
+- Fix Elasticsearch indices to work when running tests in parallel
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: This is actually not true, as old version worked with Django 1.10, which was required by Resolwe. It was changed to keep the functionality when requirement was bumped to Django 1.11. But from user"s point of view nothing has changed.

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -44,17 +50,10 @@ Fixed
...
+- Fix Elasticsearch pagination error
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Add a more descriptive entry.

Comment (tjanez) [reply docs/CHANGELOG.rst]: Done.

File: docs/CHANGELOG.rst (hunk 2 above)
Comment (tjanez) [reply docs/CHANGELOG.rst]: Ok, I see. I've removed the CHANGELOG.

I would ask you to put this info to the corresponding commit message next time, as it helps understand why the change was made.

File: docs/CHANGELOG.rst (hunk 1 above)
Comment (tjanez) [reply docs/CHANGELOG.rst]: I've reverted the change and added information about all `3.x` releases that happened in the mean time.

Comment (dblenkus) [reply docs/CHANGELOG.rst]: Thanks!

**Comment by tjanez** [comment]
The Travis builds timed out due to [an issue they have with pulling Docker images from Docker Hub](https://twitter.com/traviscistatus/status/923123856948129792).
